### PR TITLE
chore: Integrate rules_lint and format files

### PR DESCRIPTION
**Pull request by mrmeku**
---

### Changes are visible to end-users: no

### CODE REVIEW

**Comment by mrmeku** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/aspect-build/rules_cypress/78?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#78** <a href="https://app.graphite.dev/github/pr/aspect-build/rules_cypress/78?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#76** <a href="https://app.graphite.dev/github/pr/aspect-build/rules_cypress/76?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#77](https://github.com/aspect-build/rules_cypress/pull/77) <a href="https://app.graphite.dev/github/pr/aspect-build/rules_cypress/77?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mrmeku and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: MODULE.bazel [hunk 1]
```diff
@@ -12,7 +12,7 @@ bazel_dep(name = "bazel_skylib", version = "1.4.1")
 bazel_dep(name = "platforms", version = "0.0.5")
 bazel_dep(name = "rules_nodejs", version = "5.8.2")
 
-node = use_extension("@rules_nodejs//nodejs:extensions.bzl", "node")
+node = use_extension("@rules_nodejs//nodejs:extensions.bzl", "node", dev_dependency = True)
```
Comment (gregmagolan) [inline MODULE.bazel:15]: push this down bellow the `####### Dev dependencies ########` cutline

File: .github/workflows/buildifier.yaml [hunk 2]
```diff
@@ -1,16 +1,13 @@
 name: Buildifier
-
```
Comment (gregmagolan) [inline .github/workflows/buildifier.yaml:2]: yaml formatter tool out the whitespace?

Comment (mrmeku) [reply .github/workflows/buildifier.yaml:2]: Yep, that’s what prettier runs on by default with rules lint apparently. Maybe we should file a rules lint bug?

File: MODULE.bazel (hunk 1 above)
Comment (gregmagolan) [reply MODULE.bazel:16]: I'll pre-factor this out into a different PR as it is a load bearing change

File: .github/workflows/buildifier.yaml (hunk 2 above)
Comment (gregmagolan) [reply .github/workflows/buildifier.yaml]: Indeed. I rebased this and tried running `bazel run //:format` and it destroys all of the whitespace in the config yaml even tho prettier is configured for JavaScript only: `javascript = ":prettier",`.

@alexeagle Do you know what is going on here?

**Review by gregmagolan — CHANGES_REQUESTED** [review]
Need to sort out why prettier is formatting more than just JavaScript here. Probably a rules_lint bug.

File: .github/workflows/buildifier.yaml (hunk 2 above)
Comment (gregmagolan) [reply .github/workflows/buildifier.yaml]: Hmm. This PR only has the js formatter configured:

```
prettier.prettier_binary(
    name = "prettier",
    # Allow the binary to be run outside bazel
    env = {"BAZEL_BINDIR": "."},
)

format_multirun(
    name = "format",
    javascript = ":prettier",
)
```

but it is formatting JavaScript, JSON, TypeScript, Shell & YAML.... are those other formatters unintentionally on by default or is prettier formatting them?

Here is from my local command line:

```
$ git status
On branch mrmeku-rules_lint
Your branch is up to date with 'origin/mrmeku-rules_lint'.

$ bazel run //:format
INFO: Analyzed target //:format (96 packages loaded, 5352 targets configured).
INFO: Found 1 target...
Target //tools/format:format up-to-date:
  bazel-bin/tools/format/format.bash
INFO: Elapsed time: 0.410s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/tools/format/format.bash
e2e/workspace/module_test/cypress.config.js 23ms (unchanged)
e2e/workspace/module_test/module_test.cy.js 1ms (unchanged)
e2e/workspace/module_test/runner.js 3ms (unchanged)
e2e/workspace/server_example/cypress.config.js 7ms (unchanged)
e2e/workspace/server_example/server.js 2ms (unchanged)
e2e/workspace/server_example/server_example_test.cy.js 1ms (unchanged)
Formatted JavaScript in 0m0.114s
.bcr/metadata.template.json 17ms (unchanged)
.bcr/source.template.json 0ms (unchanged)
e2e/workspace/cli_test/tsconfig.json 1ms (unchanged)
e2e/workspace/package.json 1ms (unchanged)
package.json 0ms (unchanged)
renovate.json 1ms (unchanged)
Formatted JSON in 0m0.084s
e2e/workspace/cli_test/cli_test.cy.ts 81ms (unchanged)
e2e/workspace/cli_test/cypress.config.ts 6ms (unchanged)
Formatted TypeScript in 0m0.161s
Formatted Shell in 0m0.001s
Formatted YAML in 0m0.007s

$ git status
On branch mrmeku-rules_lint
Your branch is up to date with 'origin/mrmeku-rules_lint'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   .github/workflows/buildifier.yaml
	modified:   .github/workflows/ci.yaml
	modified:   .github/workflows/release.yml
	modified:   .pre-commit-config.yaml
	modified:   e2e/workspace/pnpm-lock.yaml
	modified:   pnpm-lock.yaml
```

[Prettier 1.14: YAML Support](https://prettier.io/blog/2018/07/29/1.14.0)

Comment (gregmagolan) [reply .github/workflows/buildifier.yaml]: Using rules_lint 0.19.0 in this PR:
```
bazel_dep(name = "aspect_rules_lint", version = "0.19.0", dev_dependency = True)
```

Comment (alexeagle) [reply .github/workflows/buildifier.yaml]: rules_lint requires explicit opt-out for tools that are builtin.

`yaml = None` should turn off the formatting

**Review by gregmagolan — APPROVED** [review]

**Pull request merged**